packages/cosmic-swingset/src/agcc-shim.js

```diff
@@ -0,0 +1,21 @@
+/* global process */
+import { Fail } from '@agoric/assert';
+
+let agccP;
+const { AGVM_FROM_AGD, AGVM_TO_AGD } = process.env;
+if (AGVM_FROM_AGD && AGVM_TO_AGD) {
+  const fd = [parseInt(AGVM_FROM_AGD, 10), parseInt(AGVM_TO_AGD, 10)];
+  for (const f of fd) {
+    Number.isSafeInteger(f) || Fail(`fd ${f} must be a safe integer`);
+    f >= 0 || Fail(`fd ${f} must be non-negative`);
```

Comment:
The supported range should be far smaller; [POSIX only guarantees up to 9](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_07) and bash tops out at 1023 (at least with default configuration).
```suggestion
    (Number.isInteger(f) && f >= 0 && f <= 9) ||
      Fail`fd ${f} must be an integer from 0 to 9`;

```